packages/@o3r/core/schematics/application/index.ts

```diff
@@ -0,0 +1,76 @@
+import { apply, chain, externalSchematic, MergeStrategy, mergeWith, move, renameTemplateFiles, Rule, Schematic, schematic, strings, template, url } from '@angular-devkit/schematics';
+import { getPackagesBaseRootFolder, getWorkspaceConfig, isNxContext } from '@o3r/schematics';
+import * as path from 'node:path';
+import type { NgGenerateApplicationSchema } from './schema';
+import type { PackageJson } from 'type-fest';
+
+/**
+ * Add an Otter application to a monorepo
+ *
+ * @param options Schematic options
+ */
+export function generateApplication(options: NgGenerateApplicationSchema): Rule {
+  const cleanName = strings.dasherize(options.name);
+
+  const addProjectSpecificFiles = (targetPath: string, rootDependencies: Record<string, string | undefined>) => {
+
+    return mergeWith(apply(url('./templates'), [
+      template({
+        ...options,
+        name: cleanName,
+        rootDependencies
+      }),
+      move(targetPath),
+      renameTemplateFiles()
+    ]), MergeStrategy.Overwrite);
+  };
+
+  const ngCliUpdate: Rule = (tree, context) => {
+
+    if (options.style) {
+      context.logger.warn('The provided "style" option will be ignored. "SCSS" will be used.');
+    }
+
+    const config = getWorkspaceConfig(tree);
+    if (!config) {
+      throw new Error('No workspace configuration file found');
+    }
+    if (isNxContext(tree)) {
+      throw new Error('NX is not currently supported. Feature tracked via https://github.com/AmadeusITGroup/otter/issues/720');
+    }
+
+    const defaultRoot = getPackagesBaseRootFolder(tree, context, config, 'application');
+
+    /** Path to the folder where to generate the new application */
+    const targetPath = path.posix.join('/', options.path || defaultRoot, cleanName);
+    const extendedOptions = { ...options, targetPath, name: cleanName, projectName: cleanName };
+
+    const rootPackageJson = tree.readJson('/package.json') as PackageJson;
+
+    const rootDependencies = { ...rootPackageJson.dependencies, ...rootPackageJson.devDependencies };
+
+    const angularAppSchema = context.engine.createCollection('@schematics/angular').createSchematic('application');
+
+    const getOptions = (schema: Schematic<any, any>) => {
```

Comment:
https://github.com/AmadeusITGroup/otter/blob/3d2b71b64dba8a27bf6beffe8889fcbe9dc6e197/packages/%40o3r/schematics/src/rule-factories/ng-add/index.ts#L58 

Is it possible to use the same method for those two cases ?

Reply:
Not really, the `ng-add` method does the filtering, here we simply return the array of keys.

I could factorize to get the schema options but I think it's overkill: 
```
const getSchemaOptions = (schema: Schematic<any, any>) => {
  return schema.description.schemaJson?.properties || {};
};
```